src/Microsoft.AspNetCore.Mvc.TagHelpers/AnchorTagHelper.cs

```diff
@@ -180,6 +191,18 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
                     }
                 }
 
+                if (Area != null)
+                {
+                    if (routeValues == null)
+                    {
+                        routeValues = new Dictionary<string, object>(StringComparer.OrdinalIgnoreCase);
+                    }
+
+                    // If a user specifies asp-area and asp-route-area we want the first one to always win
+                    // so we unconditionally replace the value in the dictionary with that one.
```

Comment:
So many ones. Easy to get lost halfway through. Could just say `// Unconditionally replace any value from asp-route-area.` and be done.